exporter/snowflakeexporter/internal/utility/utility.go

```diff
@@ -0,0 +1,102 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package utility provides utility functions for the snowflakeexporter package to consolidate code
+package utility // "github.com/observiq/bindplane-agent/exporter/snowflakeexporter/internal/utility"
+
+import (
+	"database/sql/driver"
+	"encoding/hex"
+	"encoding/json"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+)
+
+// Array implements driver.Valuer interface allowing arrays to be sent to Snowflake
+type Array []any
+
+// Value marshals the underlying array and then casts to a string value so it can be viewed in Snowflake
+func (a Array) Value() (driver.Value, error) {
+	if a == nil {
+		return nil, nil
+	}
+	bytes, err := json.Marshal(a)
+	if err != nil {
+		return nil, err
+	}
+	return string(bytes), nil
+}
+
+// ConvertAttributesToString converts the pcommon.Map into a JSON string representation
+// this is due to a bug/lacking feature with the snowflake driver that prevents maps from being inserted into VARIANT & OBJECT columns
+// github issue: https://github.com/snowflakedb/gosnowflake/issues/217
+func ConvertAttributesToString(attributes pcommon.Map, logger *zap.Logger) string {
+	m := make(map[string]string, attributes.Len())
+	attributes.Range(func(k string, v pcommon.Value) bool {
+		m[k] = v.AsString()
+		return true
+	})
```

Comment:
Any reason we can't just marshal attributes.AsRaw()?
Otherwise this might look weird with nested attribute maps.

Reply:
Yea I don't see why not, I should've noticed that while adding that new Value func